### PR TITLE
Add `using.macro`

### PR DESCRIPTION
**Pull request by Veetaha**
Hi!

### CODE REVIEW

**Comment by jgierer12** [comment]
Wow, that's a great one!

File: readme.md [hunk 1]
```diff
@@ -43,6 +43,9 @@
 - [`@ts-delight/switch-expr.macro`](https://www.npmjs.com/package/@ts-delight/switch-expr.macro): An expression-oriented fluent alternative to javascript's switch-statement that compiles away to ternary expressions
 - [`@ts-delight/async-to-generator.macro`](https://www.npmjs.com/package/@ts-delight/async-to-generator.macro): Transform async functions to generator functions
 
+### Object lifetime / memory management
+- [`using.macro`](https://github.com/Veetaha/using.macro): Dispose local handles automatically with C# - like [`using` declaration](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/using) in TypeScript/JavaScript.
```
Comment (jgierer12) [inline readme.md]: Do you think there would be a better way to phrase this, maybe similarly to your repo's description? I feel like this doesn't convey a good sense of what the macro does.

Comment (Veetaha) [reply readme.md]: `Call objects destructors on scope escape with automatic try-finally blocks`?
I suppose the sentence should start with a verb...

Comment (Veetaha) [reply readme.md]: @jgierer12 Well, I tried my best to rephrase (taking into account that I am not fluent in English). If you have better phrasing, please, suggest!

Comment (jgierer12) [reply readme.md]: Thanks, I think that works well.

File: readme.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 - [Macros](#macros)
   - [General](#general)
+  - [Object lifetime / memory management](#object-lifetime-/-memory-management)
```
Comment (jgierer12) [inline readme.md]:
```suggestion
  - [Object lifetime / memory management](#object-lifetime--memory-management)
```

**Review by jgierer12 — CHANGES_REQUESTED** [review]

**Comment by jgierer12** [comment]
Thanks! :tada:

**Pull request merged**